src/peft/tuners/lora/gptq.py

```diff
@@ -65,8 +70,9 @@ def forward(self, x: torch.Tensor):
             return result
 
         for active_adapter in self.active_adapters:
-            if active_adapter not in self.lora_A.keys():
+            if not self._adapter_in_lora_keys(active_adapter):
                 continue
```

Comment:
@SunMarc  Found another micro optimization opportunity. In the loop in `forward` it is allocating an iterable `keys` every single time. I don't see anywhere that updates the `self.lora_A` so I adde a `lru_cache` to remove this allocation after first foward pass.

Reply:
Yes, within the `forward` call, this should be safe to do. I wouldn't expect any measurable speed up, but it shouldn't hurt.

Reply:
I have already removed the lru cache for all instance and just moved the allocation once outside the loop.